### PR TITLE
Added tests to show errors in django 1.10

### PR DESCRIPTION
**Pull request by jbzdak**
Adding this PR to show bug in `django-enumfield` in the newly released `Django==1.10`. 

Problem was (most probably) caused by this changeset in Django repository: https://github.com/django/django/commit/7f51876f99851fdc3fef63aecdfbcffa199c26b9, and especially by the change to `get_deferred_fields`.

Let's consider this model: 

```
class Person(models.Model):
    example = models.CharField(max_length=100, default="foo")
    status = EnumField(PersonStatus, default=PersonStatus.ALIVE)
```

`EnumField` users Python properties under the hood, that is: `Person.status` is a property, this causes check used by `get_deferred_fields` to always treat `status` as a deffered field (that is always `"status" not in self.__dict__`). This causes all kinds of fun, for example excludes status from fields that are updated.

### CODE REVIEW

**Comment by listingmirror** [comment]
Ah this is perhaps why in a brand new model with EnumFIeld... .save() blows up without a force_insert=True....

File: .travis.yml [hunk 1]
```diff
@@ -12,12 +12,15 @@ env:
...
   - DJANGO=1.8
-  - DJANGO==1.9rc1
+  - DJANGO=1.9
+  - DJANGO=1.10
 
 matrix:
   exclude:
     - python: 3.3
       env: DJANGO==1.9rc1
+    - python: 3.3
```
Comment (listingmirror) [inline .travis.yml]: Shouldn't DJango 1.10 be tested with Python 3.5? That seems the most common combo

**Comment by jbzdak** [comment]
@listingmirror I beileve so --- I also got exception on ``save()`` --- in this case cause was as follows: 

* save is called
* EnumField is marked as deffered
* save method set update_fields automatically (update_fields = all_fields - deferred_fields)
* since we have update_fields save assumes that we are updating not inserting
* We are updating and there is not pk set --- exception. 

This **might** be a bug in Django, but it is triggered by ``get_deferred_fields``. 

If you need a working version: here is a version that works: https://github.com/i2biz/django-enumfield/tree/jbzdak/django110 (but please check the tests as some tests needed to be skipped due to changes I had to make).

**Comment by listingmirror** [comment]
@jbzdak Based on no issues in Django issue tracker around save, and this is the only plugin breaking like this, I am going to make the claim the bug is here not in Django.

With that said, what is the right thing to do? I can spare some cycles to put together a PR.. but also debating just ripping this plugin out and doing something else. I don't personally care about any of the state transition stuff, just want a simple way to store enums without a separate database table to join to.

**Comment by mcgeeco** [comment]
@listingmirror I wouldn't be so sure - found at least one other project affected: https://github.com/jpwatts/django-positions/issues/49

**Comment by Swamii** [comment]
@jbzdak Thanks for your help. This test was included with #45 (previously #43).

**Pull request closed without merging**